gateway/tests/e2e/providers/anthropic.rs

```diff
@@ -1,37 +1,57 @@
+use std::collections::HashMap;
+
 use crate::providers::common::{E2ETestProvider, E2ETestProviders};
 
 crate::generate_provider_tests!(get_providers);
 crate::generate_batch_inference_tests!(get_providers);
 
 async fn get_providers() -> E2ETestProviders {
+    let mut map = HashMap::new();
+    if let Ok(api_key) = std::env::var("ANTHROPIC_API_KEY") {
+        map.insert("ANTHROPIC_API_KEY".to_string(), api_key);
+    }
+
+    let credentials = if map.is_empty() { None } else { Some(map) };
+
```

Comment:
I'll not repeat this comment per-provider but let's make this map empty if there's no env var. I'd use some code like 

```rust
let credentials = match std::env::var("ANTHROPIC_API_KEY") {
    Some(key) => HashMap::from([("anthropic_api_key", key)]),
    None => HashMap::new(),
};
```